app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java

```diff
@@ -489,15 +503,41 @@ private static void handlegetHistoricDataResponse(byte[] buffer) {
             m_communicationStarted = false;
 
             Log.i(TAG, "Full data that was received is " + HexDump.dumpHexString(m_full_data));
-            LibreBlock.createAndSave("blukon", now, m_full_data, 0);
-            
-            Intent intent = new Intent(Intents.XDRIP_PLUS_LIBRE_DATA);
-            Bundle bundle = new Bundle();
-            bundle.putByteArray(Intents.LIBRE_DATA_BUFFER, m_full_data);
-            bundle.putLong(Intents.LIBRE_DATA_TIMESTAMP, now);
-            intent.putExtras(bundle);
-            intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);
-            xdrip.getAppContext().sendBroadcast(intent);
+
```

Comment:
Can you replace this code block with NFCReaderX.HandleGoodReading and verify that it works with the oopalgorithm or without it?

Reply:
Thanks, that's much neater. Hadn't wanted to make changes to the LibreOOPAlgorithm code that were too significant, but it seems to work just fine. I'm just leaving it overnight to do some history testing for both versions; will submit the corrected version tomorrow morning.